packages/components/src/custom-select-control-v2/types.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import type * as Ariakit from '@ariakit/react';
+
+export type CustomSelectContext =
+	| {
+			/**
+			 * The store object returned by Ariakit's `useSelectStore` hook.
+			 */
+			store: Ariakit.SelectStore;
+	  }
+	| undefined;
+
+export type CustomSelectProps = {
+	/**
+	 * The child elements. This should be composed of CustomSelect.Item components.
```

Comment:
Another (low-priority) follow-up would be to add some sort of check in `CustomSelectControl` to make sure that the children are effectively `CustomSelect.Item` components